components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/notifier/AbstractNotifier.java

```diff
@@ -20,24 +20,53 @@
 
 import com.google.gson.Gson;
 import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.apimgt.eventing.EventPublisher;
+import org.wso2.carbon.apimgt.eventing.EventPublisherEvent;
+import org.wso2.carbon.apimgt.eventing.EventPublisherException;
+import org.wso2.carbon.apimgt.eventing.EventPublisherFactory;
+import org.wso2.carbon.apimgt.eventing.EventPublisherType;
 import org.wso2.carbon.apimgt.impl.APIConstants;
+import org.wso2.carbon.apimgt.impl.internal.ServiceReferenceHolder;
 import org.wso2.carbon.apimgt.impl.notifier.events.Event;
+import org.wso2.carbon.apimgt.impl.notifier.exceptions.NotifierException;
 import org.wso2.carbon.apimgt.impl.utils.APIUtil;
 
 /**
  * Abstract class to handle common implementations related to notifiers
  */
 public abstract class AbstractNotifier implements Notifier {
+    private static final Log log = LogFactory.getLog(AbstractNotifier.class);
+    private EventPublisher eventPublisher;
 
-    protected void publishEventToEventHub(Event event) {
+    public AbstractNotifier() throws NotifierException {
+        // Initialize event publisher
+        try {
```

Comment:
Do this only if the feature flag is enabled

Reply:
Corrected.